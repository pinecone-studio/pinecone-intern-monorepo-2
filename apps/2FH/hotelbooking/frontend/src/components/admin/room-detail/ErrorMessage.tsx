import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { AlertCircle } from 'lucide-react';

interface ErrorMessageProps {
  message: string;
}

export const ErrorMessage = ({ message }: ErrorMessageProps) => {
  return (
    <div className="min-h-screen bg-gray-50 p-6">
      <div className="max-w-6xl mx-auto">
        <Card>
          <CardHeader>
            <CardTitle className="flex items-center gap-2 text-red-600">
              <AlertCircle className="h-5 w-5" />
              Error
            </CardTitle>
          </CardHeader>
          <CardContent>
            <p className="text-gray-700">{message}</p>
          </CardContent>
        </Card>
      </div>
    </div>
  );
};


