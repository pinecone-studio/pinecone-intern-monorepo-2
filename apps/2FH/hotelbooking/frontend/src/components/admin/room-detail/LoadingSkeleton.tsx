import { Card, CardContent, CardHeader } from '@/components/ui/card';

export const LoadingSkeleton = () => {
  return (
    <div className="min-h-screen bg-gray-50 p-6">
      <div className="max-w-6xl mx-auto">
        {/* Header Skeleton */}
        <div className="flex items-center gap-4 mb-8">
          <div className="h-10 w-20 bg-gray-200 rounded animate-pulse"></div>
          <div className="space-y-2">
            <div className="h-8 w-64 bg-gray-200 rounded animate-pulse"></div>
            <div className="h-4 w-32 bg-gray-200 rounded animate-pulse"></div>
          </div>
        </div>

        <div className="grid grid-cols-1 lg:grid-cols-2 gap-6">
          {/* Left Side Skeleton */}
          <div className="space-y-6">
            <Card>
              <CardHeader>
                <div className="h-6 w-32 bg-gray-200 rounded animate-pulse"></div>
              </CardHeader>
              <CardContent className="space-y-4">
                <div className="h-4 w-full bg-gray-200 rounded animate-pulse"></div>
                <div className="h-4 w-3/4 bg-gray-200 rounded animate-pulse"></div>
                <div className="h-4 w-1/2 bg-gray-200 rounded animate-pulse"></div>
              </CardContent>
            </Card>

            <Card>
              <CardHeader>
                <div className="h-6 w-40 bg-gray-200 rounded animate-pulse"></div>
              </CardHeader>
              <CardContent className="space-y-4">
                <div className="h-4 w-full bg-gray-200 rounded animate-pulse"></div>
                <div className="h-4 w-2/3 bg-gray-200 rounded animate-pulse"></div>
              </CardContent>
            </Card>

            <Card>
              <CardHeader>
                <div className="h-6 w-36 bg-gray-200 rounded animate-pulse"></div>
              </CardHeader>
              <CardContent>
                <div className="grid grid-cols-2 gap-4">
                  <div className="h-20 bg-gray-200 rounded animate-pulse"></div>
                  <div className="h-20 bg-gray-200 rounded animate-pulse"></div>
                </div>
              </CardContent>
            </Card>
          </div>

          {/* Right Side Skeleton */}
          <div className="space-y-6">
            <Card>
              <CardHeader>
                <div className="h-6 w-32 bg-gray-200 rounded animate-pulse"></div>
              </CardHeader>
              <CardContent>
                <div className="h-64 bg-gray-200 rounded animate-pulse"></div>
              </CardContent>
            </Card>
          </div>
        </div>
      </div>
    </div>
  );
};


