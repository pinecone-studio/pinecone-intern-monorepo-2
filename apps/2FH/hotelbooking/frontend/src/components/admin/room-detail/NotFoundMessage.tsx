/* eslint-disable  */
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { Search } from 'lucide-react';

export const NotFoundMessage = () => {
  return (
    <div className="min-h-screen bg-gray-50 p-6">
      <div className="max-w-6xl mx-auto">
        <Card>
          <CardHeader>
            <CardTitle className="flex items-center gap-2 text-gray-600">
              <Search className="h-5 w-5" />
              Room Not Found
            </CardTitle>
          </CardHeader>
          <CardContent>
            <p className="text-gray-700">The room you're looking for could not be found.</p>
          </CardContent>
        </Card>
      </div>
    </div>
  );
};
