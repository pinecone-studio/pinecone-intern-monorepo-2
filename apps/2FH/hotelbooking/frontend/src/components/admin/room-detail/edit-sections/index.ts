export { BasicInfoSection } from './BasicInfoSection';
export { AmenitiesSection } from './AmenitiesSection';
export { ImagesSection } from './ImagesSection';
export { DetailsSection } from './DetailsSection';


