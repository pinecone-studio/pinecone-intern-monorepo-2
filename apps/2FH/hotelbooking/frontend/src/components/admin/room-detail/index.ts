export { RoomInfoCard } from './RoomInfoCard';
export { RoomAmenitiesCard } from './RoomAmenitiesCard';
export { RoomImagesCard } from './RoomImagesCard';
export { LoadingSkeleton } from './LoadingSkeleton';
export { ErrorMessage } from './ErrorMessage';
export { NotFoundMessage } from './NotFoundMessage';
export { RoomDetailPage } from './RoomDetailPage';
export { EditRoomModal } from './EditRoomModal';
