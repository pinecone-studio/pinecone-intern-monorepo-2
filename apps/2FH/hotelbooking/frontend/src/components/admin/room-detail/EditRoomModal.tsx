/* eslint-disable  */
'use client';
import { useState } from 'react';
import { useUpdateRoomMutation } from '@/generated';
import { Button } from '@/components/ui/button';
import { Dialog, DialogContent, DialogHeader, DialogTitle, DialogFooter } from '@/components/ui/dialog';
import { BasicInfoSection, AmenitiesSection, ImagesSection, DetailsSection } from './edit-sections';

interface EditRoomModalProps {
  room: any;
  section: 'basic' | 'amenities' | 'images' | 'details';
  isOpen: boolean;
  onOpenChange: (open: boolean) => void;
  refetch: () => Promise<any>;
  roomId: string;
}

export const EditRoomModal = ({ room, section, isOpen, onOpenChange, refetch, roomId }: EditRoomModalProps) => {
  const [updateRoom, { loading: updateLoading }] = useUpdateRoomMutation();

  const normalizeStatus = (status: any) => {
    if (!status) return 'available';
    const statusStr = status.toString().toLowerCase();
    const validStatuses = ['available', 'booked', 'cancelled', 'completed', 'pending'];
    return validStatuses.includes(statusStr) ? statusStr : 'available';
  };

  const [formData, setFormData] = useState(() => ({
    ...room,
    status: normalizeStatus(room.status),
    internet: room.internet || [],
    foodAndDrink: room.foodAndDrink || [],
    bedRoom: room.bedRoom || [],
    bathroom: room.bathroom || [],
    accessibility: room.accessibility || [],
    entertainment: room.entertainment || [],
    other: room.other || [],
    roomInformation: room.roomInformation || [],
    imageURL: room.imageURL || [],
  }));

  const handleInputChange = (field: string, value: any) => {
    setFormData((prev: any) => ({
      ...prev,
      [field]: value,
    }));
  };

  const handleSave = async () => {
    try {
      const input: any = {};

      input.bedNumber = formData.bedNumber || room.bedNumber;
      const currentStatus = formData.status || room.status;
      input.status = normalizeStatus(currentStatus);
      if (section === 'basic') {
        if (formData.name !== undefined) input.name = formData.name;
        if (formData.pricePerNight !== undefined) input.pricePerNight = formData.pricePerNight;
        if (formData.typePerson !== undefined) input.typePerson = formData.typePerson;
        if (formData.roomInformation !== undefined) input.roomInformation = formData.roomInformation;
      } else if (section === 'amenities') {
        if (formData.internet !== undefined) input.internet = formData.internet;
        if (formData.foodAndDrink !== undefined) input.foodAndDrink = formData.foodAndDrink;
        if (formData.bedRoom !== undefined) input.bedRoom = formData.bedRoom;
        if (formData.bathroom !== undefined) input.bathroom = formData.bathroom;
        if (formData.accessibility !== undefined) input.accessibility = formData.accessibility;
        if (formData.entertainment !== undefined) input.entertainment = formData.entertainment;
        if (formData.other !== undefined) input.other = formData.other;
      } else if (section === 'images') {
        if (formData.imageURL !== undefined) input.imageURL = formData.imageURL;
      } else if (section === 'details') {
        if (formData.roomInformation !== undefined) input.roomInformation = formData.roomInformation;
      }
      const result = await updateRoom({
        variables: {
          updateRoomId: roomId,
          input,
        },
      });
      if (result.data?.updateRoom) {
        await refetch();
        onOpenChange(false);
      } else {
        console.error('Update failed: No data returned');
      }
    } catch (error: any) {
      console.error('Error updating room:', error);
      console.error('Error details:', error.message);
      console.error('GraphQL errors:', error.graphQLErrors);
      console.error('Network errors:', error.networkError);
      console.error('Full error object:', error);
    }
  };
  const getSectionComponent = () => {
    let component;
    switch (section) {
      case 'basic':
        component = <BasicInfoSection room={formData} handleInputChange={handleInputChange} />;
        break;
      case 'amenities':
        component = <AmenitiesSection room={formData} handleInputChange={handleInputChange} />;
        break;
      case 'images':
        component = <ImagesSection room={formData} handleInputChange={handleInputChange} />;
        break;
      case 'details':
        component = <DetailsSection room={formData} handleInputChange={handleInputChange} />;
        break;
      default:
        component = null;
    }
    return component || null;
  };
  const getSectionTitle = () => {
    const titleMap = {
      basic: 'Basic Information',
      amenities: 'Amenities',
      images: 'Images',
      details: 'Details',
    };
    return titleMap[section] || 'Edit Room';
  };
  return (
    <Dialog open={isOpen} onOpenChange={onOpenChange}>
      <DialogContent className="max-w-4xl max-h-[90vh] overflow-y-auto" data-testid="edit-room-modal">
        <DialogHeader>
          <DialogTitle data-testid="edit-room-modal-title">Edit {getSectionTitle()}</DialogTitle>
        </DialogHeader>

        <div className="space-y-6 py-4">
          <div className="min-h-[400px]" data-testid="edit-room-modal-content">
            {getSectionComponent()}
          </div>
        </div>

        <DialogFooter className="flex justify-end gap-3 pt-4 border-t">
          <Button variant="outline" onClick={() => onOpenChange(false)} disabled={updateLoading} data-testid="edit-room-modal-cancel">
            Cancel
          </Button>
          <Button onClick={handleSave} className="bg-blue-600 hover:bg-blue-700" disabled={updateLoading} data-testid="edit-room-modal-save">
            {updateLoading ? 'Saving...' : 'Save Changes'}
          </Button>
        </DialogFooter>
      </DialogContent>
    </Dialog>
  );
};
