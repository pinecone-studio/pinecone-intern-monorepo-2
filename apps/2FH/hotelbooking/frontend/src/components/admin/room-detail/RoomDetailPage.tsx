'use client';
import { useParams, useRouter } from 'next/navigation';
import { useState } from 'react';
import { useGetRoomQuery } from '@/generated';
import { Button } from '@/components/ui/button';
import { ArrowLeft } from 'lucide-react';
import { RoomInfoCard, RoomAmenitiesCard, RoomImagesCard, LoadingSkeleton, ErrorMessage, NotFoundMessage } from '@/components/admin/room-detail';

export const RoomDetailPage = () => {
  const params = useParams();
  const router = useRouter();
  const roomId = params.id as string;

  const [editModalState, setEditModalState] = useState<{
    isOpen: boolean;
    section: 'basic' | 'amenities' | 'images' | 'details';
  }>({
    isOpen: false,
    section: 'basic',
  });

  const { data, loading, error, refetch } = useGetRoomQuery({
    variables: { getRoomId: roomId },
  });

  if (loading) {
    return <LoadingSkeleton data-cy="RoomDetail-Loading" />;
  }

  if (error) {
    return <ErrorMessage data-cy="RoomDetail-Error" message={error.message} />;
  }

  if (!data?.getRoom) {
    return <NotFoundMessage data-cy="RoomDetail-NotFound" />;
  }

  const room = {
    ...data.getRoom,
    status: 'available', // Default status since we're not fetching it from GraphQL
  };

  return (
    <div className="min-h-screen bg-gray-50 p-6" data-cy="RoomDetail-Page">
      <div className="max-w-6xl mx-auto">
        {/* Header */}
        <div className="flex items-center gap-4 mb-8" data-cy="RoomDetail-Header">
          <Button variant="outline" onClick={() => router.back()} className="flex items-center gap-2" data-cy="RoomDetail-BackButton">
            <ArrowLeft size={16} />
            Back
          </Button>
          <div data-cy="RoomDetail-Title">
            <h1 className="text-3xl font-bold text-gray-900">{room.name}</h1>
            <div className="flex items-center gap-2 text-gray-600" data-cy="RoomDetail-Price">
              <span>
                ${room.pricePerNight}/night â€¢ {room.typePerson} People
              </span>
            </div>
          </div>
        </div>

        <div className="grid grid-cols-1 lg:grid-cols-2 gap-6">
          {/* Left Side - Information Sections */}
          <div className="space-y-6">
            <RoomInfoCard data-cy="RoomInfoCard" room={room} editModalState={editModalState} setEditModalState={setEditModalState} refetch={refetch} roomId={roomId} />
            <RoomAmenitiesCard data-cy="RoomAmenitiesCard" room={room} editModalState={editModalState} setEditModalState={setEditModalState} refetch={refetch} roomId={roomId} />
          </div>

          {/* Right Side - Images */}
          <div className="space-y-6">
            <RoomImagesCard data-cy="RoomImagesCard" room={room} editModalState={editModalState} setEditModalState={setEditModalState} refetch={refetch} roomId={roomId} />
          </div>
        </div>
      </div>
    </div>
  );
};
