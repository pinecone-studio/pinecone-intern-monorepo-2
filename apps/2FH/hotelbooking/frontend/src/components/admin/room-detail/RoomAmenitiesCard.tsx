'use client';
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { Button } from '@/components/ui/button';
import { Wifi, Utensils, Bed, Accessibility, Gamepad2, Coffee } from 'lucide-react';
import { EditRoomModal } from './EditRoomModal';

interface RoomAmenitiesCardProps {
  room: any;
  editModalState: {
    isOpen: boolean;
    section: 'basic' | 'amenities' | 'images' | 'details';
  };
  setEditModalState: (_state: { isOpen: boolean; section: 'basic' | 'amenities' | 'images' | 'details' }) => void;
  refetch: () => Promise<any>;
  roomId: string;
}

export const RoomAmenitiesCard = ({ room, editModalState, setEditModalState, refetch, roomId }: RoomAmenitiesCardProps) => {
  // Create a comprehensive list of all amenities with their values and labels
  const getAllAmenities = () => {
    const amenityCategories = [
      { key: 'internet', label: 'Internet', icon: <Wifi className="h-4 w-4" /> },
      { key: 'foodAndDrink', label: 'Food & Drink', icon: <Utensils className="h-4 w-4" /> },
      { key: 'bedRoom', label: 'Bedroom', icon: <Bed className="h-4 w-4" /> },
      { key: 'bathroom', label: 'Bathroom', icon: <Coffee className="h-4 w-4" /> },
      { key: 'accessibility', label: 'Accessibility', icon: <Accessibility className="h-4 w-4" /> },
      { key: 'entertainment', label: 'Entertainment', icon: <Gamepad2 className="h-4 w-4" /> },
      { key: 'other', label: 'Other', icon: <Coffee className="h-4 w-4" /> },
    ];

    const allAmenities: any[] = [];

    amenityCategories.forEach((category) => {
      const categoryValue = room[category.key];
      if (categoryValue) {
        // Handle both single values and arrays
        const values = Array.isArray(categoryValue) ? categoryValue : [categoryValue];
        values.forEach((value: string) => {
          if (value) {
            allAmenities.push({
              key: `${category.key}-${value}`,
              category: category.label,
              value: value,
              icon: category.icon,
            });
          }
        });
      }
    });

    return allAmenities;
  };

  const amenities = getAllAmenities();

  return (
    <Card>
      <CardHeader className="flex flex-row items-center justify-between">
        <CardTitle>Room Amenities</CardTitle>
        <EditRoomModal
          room={room}
          section="amenities"
          isOpen={editModalState.isOpen && editModalState.section === 'amenities'}
          onOpenChange={(open) => setEditModalState({ isOpen: open, section: 'amenities' })}
          refetch={refetch}
          roomId={roomId}
        />
        <Button size="sm" className="bg-blue-600 hover:bg-blue-700 text-white" onClick={() => setEditModalState({ isOpen: true, section: 'amenities' })}>
          Edit
        </Button>
      </CardHeader>
      <CardContent>
        {amenities.length === 0 ? (
          <div className="text-center py-8 text-gray-500">No amenities configured for this room</div>
        ) : (
          <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
            {amenities.map((amenity) => (
              <div key={amenity.key} className="flex items-center gap-3 p-3 border rounded-lg">
                <div className="text-blue-600">{amenity.icon}</div>
                <div className="flex-1">
                  <div className="font-medium text-gray-900">{amenity.value.replace(/_/g, ' ').replace(/\b\w/g, (l: string) => l.toUpperCase())}</div>
                  <div className="text-sm text-gray-600">{amenity.category}</div>
                </div>
              </div>
            ))}
          </div>
        )}
      </CardContent>
    </Card>
  );
};
