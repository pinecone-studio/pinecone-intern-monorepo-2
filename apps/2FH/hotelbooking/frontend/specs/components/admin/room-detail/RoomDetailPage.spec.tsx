/* eslint-disable  */
import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { RoomDetailPage } from '@/components/admin/room-detail/RoomDetailPage';
import { useGetRoomQuery } from '@/generated';
import { useParams, useRouter } from 'next/navigation';

// Mock dependencies
jest.mock('@/generated', () => ({
  useGetRoomQuery: jest.fn(),
}));

jest.mock('next/navigation', () => ({
  useParams: jest.fn(),
  useRouter: jest.fn(),
}));

jest.mock('@/components/admin/room-detail', () => ({
  RoomInfoCard: ({ room, editModalState, setEditModalState, refetch, roomId }: any) => (
    <div data-testid="room-info-card">
      <div>Room Info Card</div>
      <div>Room: {room.name}</div>
      <button onClick={() => setEditModalState({ isOpen: true, section: 'basic' })}>Edit Basic Info</button>
    </div>
  ),
  RoomAmenitiesCard: ({ room, editModalState, setEditModalState, refetch, roomId }: any) => (
    <div data-testid="room-amenities-card">
      <div>Room Amenities Card</div>
      <button onClick={() => setEditModalState({ isOpen: true, section: 'amenities' })}>Edit Amenities</button>
    </div>
  ),
  RoomImagesCard: ({ room, editModalState, setEditModalState, refetch, roomId }: any) => (
    <div data-testid="room-images-card">
      <div>Room Images Card</div>
      <button onClick={() => setEditModalState({ isOpen: true, section: 'images' })}>Edit Images</button>
    </div>
  ),
  LoadingSkeleton: () => <div data-testid="loading-skeleton">Loading...</div>,
  ErrorMessage: ({ message }: any) => <div data-testid="error-message">Error: {message}</div>,
  NotFoundMessage: () => <div data-testid="not-found-message">Room not found</div>,
}));

const mockUseGetRoomQuery = useGetRoomQuery as jest.MockedFunction<typeof useGetRoomQuery>;
const mockUseParams = useParams as jest.MockedFunction<typeof useParams>;
const mockUseRouter = useRouter as jest.MockedFunction<typeof useRouter>;

describe('RoomDetailPage', () => {
  const mockRoom = {
    id: 'room-1',
    name: 'Deluxe Suite',
    pricePerNight: 150000,
    typePerson: 'double',
    bedNumber: 2,
    status: 'available',
    internet: ['free_wifi'],
    foodAndDrink: ['free_breakfast'],
    bedRoom: ['air_conditioner'],
    bathroom: ['private'],
    accessibility: [],
    entertainment: ['tv'],
    other: ['desk'],
    roomInformation: ['private_bathroom', 'air_conditioner'],
    imageURL: ['https://example.com/image1.jpg'],
  };

  const mockRouter = {
    back: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    refresh: jest.fn(),
    forward: jest.fn(),
    prefetch: jest.fn(),
  };

  const mockRefetch = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    mockUseParams.mockReturnValue({ id: 'room-1' });
    mockUseRouter.mockReturnValue(mockRouter);
  });

  it('renders loading state', () => {
    mockUseGetRoomQuery.mockReturnValue({
      data: undefined,
      loading: true,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    render(<RoomDetailPage />);

    expect(screen.getByTestId('loading-skeleton')).toBeInTheDocument();
  });

  it('renders error state', () => {
    const errorMessage = 'Failed to load room';
    mockUseGetRoomQuery.mockReturnValue({
      data: undefined,
      loading: false,
      error: { message: errorMessage } as any,
      refetch: mockRefetch,
    } as any);

    render(<RoomDetailPage />);

    expect(screen.getByTestId('error-message')).toBeInTheDocument();
    expect(screen.getByText(`Error: ${errorMessage}`)).toBeInTheDocument();
  });

  it('renders not found state when room data is null', () => {
    mockUseGetRoomQuery.mockReturnValue({
      data: { getRoom: null },
      loading: false,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    render(<RoomDetailPage />);

    expect(screen.getByTestId('not-found-message')).toBeInTheDocument();
  });

  it('renders room details when data is available', () => {
    mockUseGetRoomQuery.mockReturnValue({
      data: { getRoom: mockRoom },
      loading: false,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    render(<RoomDetailPage />);

    expect(screen.getByTestId('room-info-card')).toBeInTheDocument();
    expect(screen.getByTestId('room-amenities-card')).toBeInTheDocument();
    expect(screen.getByTestId('room-images-card')).toBeInTheDocument();
  });

  it('displays room name and price in header', () => {
    mockUseGetRoomQuery.mockReturnValue({
      data: { getRoom: mockRoom },
      loading: false,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    render(<RoomDetailPage />);

    expect(screen.getByText('Deluxe Suite')).toBeInTheDocument();
    expect(screen.getByText('$150000/night â€¢ double People')).toBeInTheDocument();
  });

  it('calls router.back when back button is clicked', () => {
    mockUseGetRoomQuery.mockReturnValue({
      data: { getRoom: mockRoom },
      loading: false,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    render(<RoomDetailPage />);

    const backButton = screen.getByText('Back');
    fireEvent.click(backButton);

    expect(mockRouter.back).toHaveBeenCalled();
  });

  it('renders with correct page structure', () => {
    mockUseGetRoomQuery.mockReturnValue({
      data: { getRoom: mockRoom },
      loading: false,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    render(<RoomDetailPage />);

    const pageContainer = screen.getByTestId('room-info-card').closest('.min-h-screen');
    expect(pageContainer).toHaveClass('bg-gray-50', 'p-6');
  });

  it('renders grid layout correctly', () => {
    mockUseGetRoomQuery.mockReturnValue({
      data: { getRoom: mockRoom },
      loading: false,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    render(<RoomDetailPage />);

    const gridContainer = document.querySelector('.grid.grid-cols-1.lg\\:grid-cols-2');
    expect(gridContainer).toBeInTheDocument();
  });

  it('passes correct props to child components', () => {
    mockUseGetRoomQuery.mockReturnValue({
      data: { getRoom: mockRoom },
      loading: false,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    render(<RoomDetailPage />);

    // Check that room info is passed to components
    expect(screen.getByText('Room: Deluxe Suite')).toBeInTheDocument();
  });

  it('handles edit modal state changes', () => {
    mockUseGetRoomQuery.mockReturnValue({
      data: { getRoom: mockRoom },
      loading: false,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    render(<RoomDetailPage />);

    // Click edit button to open modal
    const editButton = screen.getByText('Edit Basic Info');
    fireEvent.click(editButton);

    // The modal state should be updated (this is tested through the child components)
    expect(editButton).toBeInTheDocument();
  });

  it('sets default room status when not provided', () => {
    const roomWithoutStatus = { ...mockRoom, status: undefined };
    mockUseGetRoomQuery.mockReturnValue({
      data: { getRoom: roomWithoutStatus },
      loading: false,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    render(<RoomDetailPage />);

    // The component should set a default status of 'available'
    expect(screen.getByTestId('room-info-card')).toBeInTheDocument();
  });

  it('calls useGetRoomQuery with correct room ID', () => {
    mockUseGetRoomQuery.mockReturnValue({
      data: { getRoom: mockRoom },
      loading: false,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    render(<RoomDetailPage />);

    expect(mockUseGetRoomQuery).toHaveBeenCalledWith({
      variables: { getRoomId: 'room-1' },
    });
  });

  it('handles missing room ID parameter', () => {
    mockUseParams.mockReturnValue({ id: undefined });
    mockUseGetRoomQuery.mockReturnValue({
      data: undefined,
      loading: true,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    render(<RoomDetailPage />);

    expect(screen.getByTestId('loading-skeleton')).toBeInTheDocument();
  });

  it('matches snapshot for loading state', () => {
    mockUseGetRoomQuery.mockReturnValue({
      data: undefined,
      loading: true,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    const { container } = render(<RoomDetailPage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot for loaded state', () => {
    mockUseGetRoomQuery.mockReturnValue({
      data: { getRoom: mockRoom },
      loading: false,
      error: undefined,
      refetch: mockRefetch,
    } as any);

    const { container } = render(<RoomDetailPage />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
