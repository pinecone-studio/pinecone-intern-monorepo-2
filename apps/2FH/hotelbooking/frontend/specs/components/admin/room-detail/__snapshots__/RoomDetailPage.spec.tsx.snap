// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomDetailPage matches snapshot for loaded state 1`] = `
<div
  class="min-h-screen bg-gray-50 p-6"
  data-cy="RoomDetail-Page"
>
  <div
    class="max-w-6xl mx-auto"
  >
    <div
      class="flex items-center gap-4 mb-8"
      data-cy="RoomDetail-Header"
    >
      <button
        class="justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 flex items-center gap-2"
        data-cy="RoomDetail-BackButton"
      >
        <svg
          class="lucide lucide-arrow-left"
          fill="none"
          height="16"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m12 19-7-7 7-7"
          />
          <path
            d="M19 12H5"
          />
        </svg>
        Back
      </button>
      <div
        data-cy="RoomDetail-Title"
      >
        <h1
          class="text-3xl font-bold text-gray-900"
        >
          Deluxe Suite
        </h1>
        <div
          class="flex items-center gap-2 text-gray-600"
          data-cy="RoomDetail-Price"
        >
          <span>
            $
            150000
            /night â€¢ 
            double
             People
          </span>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-1 lg:grid-cols-2 gap-6"
    >
      <div
        class="space-y-6"
      >
        <div
          data-testid="room-info-card"
        >
          <div>
            Room Info Card
          </div>
          <div>
            Room: 
            Deluxe Suite
          </div>
          <button>
            Edit Basic Info
          </button>
        </div>
        <div
          data-testid="room-amenities-card"
        >
          <div>
            Room Amenities Card
          </div>
          <button>
            Edit Amenities
          </button>
        </div>
      </div>
      <div
        class="space-y-6"
      >
        <div
          data-testid="room-images-card"
        >
          <div>
            Room Images Card
          </div>
          <button>
            Edit Images
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RoomDetailPage matches snapshot for loading state 1`] = `
<div
  data-testid="loading-skeleton"
>
  Loading...
</div>
`;
